File: LCA.py
def find_LCA(root, n1, n2):
    if root is None:
        return None

    if root.data == n1 or root.data == n2:
        return root

    left_LCA = find_LCA(root.left, n1, n2)
    right_LCA = find_LCA(root.right, n1, n2)

    if left_LCA and right_LCA:
        return root

    return left_LCA if left_LCA else right_LCA

File: C/LCA.c
struct Node* FindLCA(struct Node* root, int n1, int n2) {
    if (root == NULL)
        return NULL;

    if (root->data == n1 || root->data == n2)
        return root;

    struct Node* leftLCA = FindLCA(root->left, n1, n2);
    struct Node* rightLCA = FindLCA(root->right, n1, n2);

    if (leftLCA && rightLCA)
        return root;
    
    return (leftLCA != NULL) ? leftLCA : rightLCA;
}
